src/components/Extensions/Details/DetailCard/useMasterDetail/useMasterDetail.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { randomId } from '@rmwc/base';
+import { List, ListItem } from '@rmwc/list';
+import { Typography } from '@rmwc/typography';
+import React, { useState } from 'react';
+
+import style from './useMasterDetail.module.scss';
+
+export interface MasterDetailTabConfig {
+  label: string;
+  tab: JSX.Element;
+}
+
+export interface MasterDetailConfig {
+  tabs: MasterDetailTabConfig[];
+}
+
+export function useMasterDetail(config: MasterDetailConfig) {
```

Comment:
1) could we consider moving this to the top-level components directory?
2) we need to rename this... `master` is no longer ok to use go/respectful-code